src/net/sourceforge/kolmafia/KoLmafia.java

```diff
@@ -1363,14 +1364,15 @@ private static void executeAdventureOnce(
 
       if (KoLCharacter.hasEquipped(ItemPool.get(ItemPool.DRUNKULA_WINEGLASS, 1))) {
         // The wine glass allows you to adventure while falling down drunk
-      } else if (KoLCharacter.getLimitmode() == Limitmode.SPELUNKY) {
+      } else if (Objects.equals(KoLCharacter.getLimitmode(), Limitmode.SPELUNKY)) {
```

Comment:
Why? Look at KoLCharacter.setLimitMode(String limitmode):

```
    } else if (limitmode.equals(Limitmode.SPELUNKY)) {
      KoLCharacter.limitmode = Limitmode.SPELUNKY;
    } else if (limitmode.equals(Limitmode.BATMAN)) {
      KoLCharacter.limitmode = Limitmode.BATMAN;
      BatManager.setCombatSkills();
    } else if (limitmode.equals(Limitmode.ED)) {
      KoLCharacter.limitmode = Limitmode.ED;
    } else {
      KoLCharacter.limitmode = limitmode;
    }
```

Here is Limitmode.java:
```
  // Limitmode
  public static final String SPELUNKY = "spelunky";
  public static final String BATMAN = "batman";
  public static final String ED = "edunder";
```

There are currently only 3 limitmodes. I suppose it would be nice if Limitmode was an enum, but as coded,  those final strings are effectively the equivalent. Specifically so you can use "==" with them, as we do, all over the place.